File: TravellerCreatorModels/Basic/BasicHex.cs
using TravellerCreatorModels.Interfaces;
using TravellerCreatorModels.Other;

namespace TravellerCreatorModels.Basic;

public class BasicHex: IHex
{
    public Coordinates Coordinates { get; set; }
    public List<IStarSystem> StarSystems { get; set; } = new();

    public BasicHex(Coordinates coordinates)
    {
        Coordinates = coordinates;
    }
}

File: TravellerCreator/ClientApp/src/app/sector-creator/models/basic/basic-hex.ts
import {IHex} from "../interfaces/hex";
import {Coordinates} from "../other/coordinates";
import {BasicStarSystem} from "./basic-star-system";

export class BasicHex implements IHex {
  coordinates: Coordinates;
  starSystems: BasicStarSystem[] = [];

  constructor(coordinates: Coordinates) {
    this.coordinates = coordinates;
  }
}
